pkg/controller/actuator_reconcile.go

```diff
@@ -91,6 +92,18 @@ func (a *actuator) Reconcile(ctx context.Context, _ logr.Logger, network *extens
 		}
 	}
 
+	if networkConfig.Overlay != nil && networkConfig.Overlay.Enabled {
+		if networkConfig.TunnelMode == nil || networkConfig.TunnelMode != nil && *networkConfig.TunnelMode == "disabled" {
```

Comment:
Can we use `ciliumv1alpha1.VXLan` here or simply do in inequality check for `geneve` (constant)?